addons/mail/static/src/core/common/attachment_list.js

```diff
@@ -102,4 +102,9 @@ export class AttachmentList extends Component {
     get isInChatWindowAndIsAlignedLeft() {
         return this.env.inChatWindow && !this.env.alignedRight;
     }
+
+    get showDeleteOnAttachmentCard() {
+        return this.attachment.isDeletable && (this.env.message?.hasTextContent !== false
```

Comment:
```js
this.env.message?.hasTextContent !== false
```
=>
```js
!this.env.message?.hasTextContent
```

Reply:
Hello @alexkuhn ,
We keep this condition as `!== false` because we intend to display the Delete icon when the value is `undefined` or `true`. Here is an alternative, 
```
this.attachment.isDeletable
            && (this.env.message?.hasTextContent || (this.env.inChatter && !this.env.message)
            || this.env.inComposer || this.props.attachments.length > 1)
```
Kindly provide your feedback on what can be done.
Thanks..

Reply:
Why not `!this.env.message || this.env.message?.hasTextContent` then? `env.inChatter` seems useless

Reply:
@alexkuhn ,
Thanks a lot for the help,
I have made the changes...